Made it work

FakeEmperor opened the pull request.

eprev committed:
v2

eprev committed:
v2

ivan-chashkin committed:
Update

ivan-chashkin committed:
Update

ivan-chashkin committed:
Update

Commit:
grunt qunit

Commit:
merge with v1

Commit:
удалил лишние файлы, оставшиеся после мержа

iketari committed:
new v1

iketari committed:
fix package.json

Commit:
qunit

Commit:
merge with v1

burlakilia committed:
fix

iketari committed:
Gruntfile.js update

Commit:
requirejs tests, grunt test

FakeEmperor committed:
Made it work

burlakilia committed:
Доработано задание на практике (watch, concurrent, express)

burlakilia commented:
Классная копия, но все таки давай ты нажмешь сам кнопки =) 
https://github.com/expressjs/generator/blob/master/templates/js/www

FakeEmperor committed:
Removed compass from app.use() chain

burlakilia committed:
Добавлена логика запуска сервера

KirSlayk committed:
"deleted commit"

KirSlayk committed:
Merge remote-tracking branch 'upstream/v1'

FakeEmperor committed:
Merge remote-tracking branch 'upstream/v0'

FakeEmperor committed:
:insert
lalka!!:wq
:q

KirSlayk committed:
LAAAAAAAAAAAAAAAAAAAAAAAAAL
Merge branch 'master' of https://github.com/SlaykovskiyHDNANO/2016_1_Lime-n-Pineapple

FakeEmperor committed:
Ok, merged again

KirSlayk committed:
Jesus

KirSlayk committed:
change gruntfile shell

iketari committed:
fu*k JAVA

KirSlayk committed:
Немножк изменил gruntfile, целых четыре строчки убрал, package.json тож автоматом изменился

KirSlayk committed:
Доработал gruntfile

KirSlayk committed:
index change

KirSlayk committed:
gitignore file modified

FakeEmperor committed:
Updated .gitignore

FakeEmperor committed:
Merge branch 'master' of https://github.com/SlaykovskiyHDNANO/2016_1_Lime-n-Pineapple

FakeEmperor committed:
Removed comments and lines
Removed redundant lines and comments :rooster:

FakeEmperor committed:
Merge branch 'v2' of https://github.com/frontend-park-mail-ru/frontend-upstream

FakeEmperor committed:
Fixed broken merge

FakeEmperor commented:
Поменяли, вроде бы.

FakeEmperor committed:
Fest template and "Hello, world!" functionality

[+] Button model (needs to be deleted)
[+] Router (for Hello, world)
[+] Main model
[+] Fest template for Hello, world.
[-] Removed temp files
[*] .gitignore ignores files in public_html/js/tmpl

KirSlayk committed:
slayk Change v2, models, collections

KirSlayk committed:
Conflict resolved

FakeEmperor committed:
Styles and templates

FakeEmperor committed:
templates and indexes

KirSlayk committed:
Merge branch 'master' of https://github.com/SlaykovskiyHDNANO/2016_1_Lime-n-Pineapple into slaykV2

FakeEmperor committed:
In progress

KirSlayk committed:
scoreboard view render

KirSlayk committed:
server change

FakeEmperor committed:
Merge branch 'slaykV2' of https://github.com/SlaykovskiyHDNANO/2016_1_Lime-n-Pineapple

KirSlayk committed:
btn_back with scoreboard leegheid

KirSlayk committed:
resolved merge conflict

KirSlayk committed:
not working

KirSlayk committed:
lol

KirSlayk committed:
resolved bag with requirejs

KirSlayk committed:
created login form, btnBack work nice, commit after work with collections

KirSlayk committed:
merge with slaykV2, login created, btnBack work nice, commit after i work with collections

KirSlayk committed:
scoreboard with sorted collection of players. I did it

madimp commented on public_html/index.html:
```diff
@@ -1,20 +1,27 @@
+	    <meta charset="utf-8" />
+	    <title>Whack-a-mole: Technopark Scholar Edition</title>
+	    <link rel="stylesheet" href="css/main.css" />
+	    <link rel="stylesheet" href="css/bootstrap.css" />
```
свои стили должны быть явно после библиотеки, чтобы иметь больший приоритет

madimp commented on public_html/js/router.js:
```diff
@@ -0,0 +1,90 @@
+
+            _createView: function (ViewClass, viewName){
+                var view = this[viewName];
+                if(this[viewName] === undefined) {
```
что будет, если я вью назову gameAction?
правильно. все сломается.
положите вьюхи в одно св-во, где будут только они.

madimp commented on public_html/js/router.js:
```diff
@@ -0,0 +1,90 @@
+                if (this.login) {
+                    this.login.hide();
+                }
+                if (this.game){
```
если у вас есть в одном месте все вьюхи, то можно циклом все скрыть и не придется править эту функцию при добавлении новой вью

madimp commented on public_html/js/views/block/btn_back.js:
```diff
@@ -0,0 +1,63 @@
+        },
+
+        render: function () {
+            this.$el.appendTo($("#view__btn_back"));
```
вью не должна сама определять, куда ей отрендерится. этим должен управлять тот код, который создает/рендерит вьюху

madimp commented on public_html/js/views/block/btn_back.js:
```diff
@@ -0,0 +1,63 @@
+            this.stopBack();
+                this.on("back", function () {
+                    console.log("[Btn_Back]: back() called");
+                    Backbone.history.history.back();
```
не проще ли в clickBtn сразу делать реакцию на клик?
либо я не понимаю, для чего вам каждый раз вешать и снимать обработчики некого back, либо вы не понимаете как работает events

madimp commented on public_html/js/views/login.js:
```diff
@@ -0,0 +1,36 @@
+        },
+
+        render: function () {
+            this.$el.appendTo($("#view__holder"));
```
аналогично прерыдуще вью

madimp commented on public_html/js/views/main.js:
```diff
@@ -0,0 +1,46 @@
+        },
+
+        render: function () {
+            this.$el.appendTo($("#view__holder"));
```
аналогично предыдущему

madimp commented on public_html/js/views/scoreboard.js:
```diff
@@ -0,0 +1,51 @@
+        },
+
+        render: function () {
+            this.$el.appendTo($("#view__holder"));
```
+1

madimp commented on public_html/js/views/scoreboard.js:
```diff
@@ -0,0 +1,51 @@
+        initialize: function () {
+            this.collection = new Scores();
+            var i;
+            for (i = 0; i < 3; i++) {
```
for (var i = 0; i < 3; i++) {
так тоже работает

KirSlayk replied:
JSLint ругается

madimp replied:
ок

madimp commented on templates/login.xml:
```diff
@@ -1,3 +1,52 @@
+<?xml version="1.0"?>
 <fest:template xmlns:fest="http://fest.mail.ru">
-    <div id="login"></div>
+        <div id="loginbox" style="margin-top:3.7vh;" class="mainbox center-block text-center">
```
margin-top:3.7vh;
стили нужно писать в таблице стилей для конкретного блока

KirSlayk replied:
почему-то xml не хавает подключение стилей через xml-stylesheet, я поднимаю белый флаг, два часа угробил

madimp commented on templates/login.xml:
```diff
@@ -1,3 +1,52 @@
+                    <!--<div style="float:right; font-size: 80%; position: relative; top:-10px"><a href="#">Forgot password?</a></div>-->
+                </div>
+
+                <div style="padding-top:30px" class="panel-body" >
```
+1

madimp commented on templates/login.xml:
```diff
@@ -1,3 +1,52 @@
+
+                <div style="padding-top:30px" class="panel-body" >
+
+                    <div style="display:none" id="login-alert" class="alert alert-danger col-sm-12"></div>
```
а вот такие стили ок

madimp commented on templates/login.xml:
```diff
@@ -1,3 +1,52 @@
+
+                    <form id="loginform" class="form-horizontal" role="form">
+
+                        <div style="margin-bottom: 25px" class="input-group">
```
+1

madimp commented on templates/login.xml:
```diff
@@ -1,3 +1,52 @@
+                            <input id="login-username" type="text" class="form-control" name="username" value="" placeholder="username or email"/>
+                        </div>
+
+                        <div style="margin-bottom: 25px" class="input-group">
```
+1

madimp commented on templates/login.xml:
```diff
@@ -1,3 +1,52 @@
+                            <input id="login-password" type="password" class="form-control" name="password" placeholder="password"/>
+                        </div>
+
+                        <div style="margin-top:10px" class="form-group">
```
+1

madimp commented on templates/login.xml:
```diff
@@ -1,3 +1,52 @@
+                                <div style="border-top: 1px solid#888; padding-top:10px; font-size:85%" >
+                                    Don't have an account!
+                                    <fest:space></fest:space>
+                                    <a href="/registration" onClick="$('#loginbox').hide(); $('#signupbox').show()">
```
js в шаблонах быть не должно. логику надо описывать в соответствующей вью

madimp commented on templates/login.xml:
```diff
@@ -1,3 +1,52 @@
+
+                        <div class="form-group">
+                            <div class="col-md-12 control">
+                                <div style="border-top: 1px solid#888; padding-top:10px; font-size:85%" >
```
+1

madimp commented on templates/login.xml:
```diff
@@ -1,3 +1,52 @@
+                        <div style="margin-top:10px" class="form-group">
+
+                            <div class="col-sm-12 controls">
+                                <a id="btn-login" href="#" class="btn btn-success" style="text-decoration: none">Login  </a>
```
кнопка сабмита должна быть кнопкой, а не ссылкой

madimp commented on templates/main_page.xml:
```diff
@@ -0,0 +1,27 @@
+<?xml version="1.0"?>
+<fest:template xmlns:fest="http://fest.mail.ru">
+        <div id="menu" class="center-block" style="width: 40%"> <!-- Блок меню -->
```
стили в .css

madimp commented on templates/scoreboard.xml:
```diff
@@ -1,3 +1,16 @@
+            <!--<ul style="margin-left:37%">-->
+            <ul class="col-lg-12 col-lg-offset-5 col-md-12 col-md-offset-5 col-sm-12 col-sm-offset-5 col-xs-12 col-xs-offset-4">
+                <fest:for iterate="json.person" index="i">
+                    <fest:space/><fest:element name="li"><fest:value>json.person[json.person.length - i - 1].name</fest:value><fest:space/><fest:value>json.person[json.person.length - i - 1].score</fest:value></fest:element>
```
json.person.length - i - 1 в чем смысл?
перенесите сортировку в сортировку коллекции. шаблон должен выводить данные как ему выдали

KirSlayk committed:
added bootstram-theme.css

KirSlayk committed:
исправлены баги из pull-request

FakeEmperor committed:
Made styles work with login.xml. See fest:insert

KirSlayk committed:
Merge pull request #1 from SlaykovskiyHDNANO/pineapple_working

Made styles work with login.xml. See fest:insert
